### DISCUSSION TITLE
Gauge with springboot - how can we debug it with breakpoints?

### DISCUSSION THREAD

**Discussion by saritsegal** [getgauge/gauge#2009] [General]
we have some gauge specs running successfully with sprint-boot annotation. We need to debug them. Breakpoint are not active and execution is not stopped. please, advice we should be done

**Comment by zabil** [comment]
Which IDE are you using? How are you setting the breakpoints or launching the debug mode? Can you also share more info on versions and how this project is set up (maven or gradle)

**Reply by saritsegal** [reply]
IDE is intelliJ

Gauge version: 1.1.8
Commit Hash: b1501f4

Plugins
-------
html-report (4.0.12)
java (0.7.15)
screenshot (0.0.1)

Springboot project with Maven

Gauge is loaded using RegisterIOC that add the following TestConfiguration to app context as follow:
applicationContext = new AnnotationConfigApplicationContext(TestConfiguration.class);

and TestConfiguration initiates StepImplementation (which contains the spec implmentation) as follow:
@Configuration
@PropertySource(value = {"classpath:application.properties"})
@ComponentScan(basePackages = "application")
public class TestConfiguration {

    @Bean
    public StepImplementation TestSteps() {
        return new StepImplementation();
    }
}

I tried running it with "Gauge Execution" configuration and with "Maven" configuration
 but in both cases - breakpoints are not active

**Comment by zabil** [comment]
> I tried running it with "Gauge Execution" configuration and with "Maven" configuration
but in both cases - breakpoints are not active

Can you explain this a bit more? The debug will only work within the IDE. You must use the test runner within IntelliJ to run it instead of running it via Maven.

**Reply by saritsegal** [reply]
what is the "test runner" within intelliJ?
I tried "Select run/debuf configuration": from the intelliJ menu:
![image](https://user-images.githubusercontent.com/19801918/119647792-5c83a500-be29-11eb-9009-08e22f7e5022.png)

**Reply by zabil** [reply]
![Kapture 2021-05-26 at 13 53 08](https://user-images.githubusercontent.com/54427/119663222-111dc680-be2a-11eb-8cd5-ecdb93695af9.gif)


Here's an example on debug working on a simple gauge maven project created with 

```
gauge init java_maven
```

If this is not working for your spring project, please share a sample project (with a simple test case) that replicates the issue as a github repository.

Alternatively Jetbrains maintains this plugin you can raise an issue here https://youtrack.jetbrains.com/issues/IDEA
